app/src/screens/ProposalDetail.js

```diff
@@ -48,43 +59,116 @@ function ProposalDetail({ proposal, onBack, requestToken }) {
     threshold,
   } = proposal
 
-  const myStakes = stakes.filter(({ entity }) =>
-    addressesEqual(entity, connectedAccount)
+  const myStake = useMemo(
+    () =>
+      stakes.find(({ entity }) => addressesEqual(entity, connectedAccount)) || {
+        amount: new BigNumber('0'),
+      },
+    [stakes, connectedAccount]
+  )
+
+  const myStakeAmountFormatted = useMemo(() => {
+    if (!myStake) {
+      return '0'
+    }
+    return formatTokenAmount(myStake.amount, stakeToken.tokenDecimals)
+  }, [stakeToken.tokenDecimals, myStake])
+
+  const totalStaked = useAccountTotalStaked()
+
+  const nonStakedTokens = useMemo(
+    () => stakeToken.balance.minus(totalStaked).plus(myStake.amount),
+    [myStake.amount, stakeToken.balance, totalStaked]
   )
-  const didIStaked = myStakes.length > 0 && [...myStakes].pop().tokensStaked > 0
+
+  const formattedMaxAvailableAmount = useMemo(() => {
+    if (!stakeToken) {
+      return '0'
+    }
+    return formatTokenAmount(nonStakedTokens, stakeToken.tokenDecimals)
+  }, [stakeToken, nonStakedTokens])
+
+  const rounding = useMemo(() => {
+    return Math.min(MAX_INPUT_DECIMAL_BASE, stakeToken.decimals)
+  }, [stakeToken])
+
```

Comment:
I wouldn't use a `useMemo` in this case where it's a basic operation. I think the cost of using it is more expensive than  for what we are trying to optimize for.